### DISCUSSION TITLE
Clarification on decision to not integrate a weather radar

### DISCUSSION THREAD

**Discussion by LinuxinaBit** [breezy-weather/breezy-weather#1515] [General]
Reading [RADAR.md](https://github.com/breezy-weather/breezy-weather/blob/main/docs/RADAR.md) I see there is reasoning for not including a radar viewer; namely the fact that there are other freely available services which can be used to access that information (i.e. various different weather radar websites that can be added as a PWA).

I find a few problems with that notion, the first being that Breezy Weather displays nothing but freely available weather data that _also_ can be accessed though other sources, and I can’t really figure out what makes displaying weather radar data _specifically_ any different.

I _really like_ this project (it’s so beautiful!), and want to have it as the only weather app on my phone, but not having a radar makes it pretty unusable in that context.
I don’t _want_ to just use a website to access a weather radar, I want an integrated app that fulfills all my weather checking needs.

This actually seems to be a common request, as there have been many others before me asking for the exact same thing:
#23 
#767 
#903 
#1178 
#1187 
#1288 
They are all well aware of other radar viewing services being available, but that’s not what these are requests for.
There is a genuine need for a fully integrated, one-stop-shop FOSS weather app, and if Breezy Weather added a radar, I think it would finally be able to provide that.

Thank you for considering this feature,
Aaron C.

P.S. If any integration examples are needed, [QuickWeather](https://f-droid.org/packages/com.ominous.quickweathe/) is a great example for how to do it.
It integrates RainViewer, but that’s ok because the real value in adding a weather radar is not that the data is somehow unique, it’s the fact that it is integrated in the app _at all_.

**Comment by LinuxinaBit** [comment]
Also note that just because coverage isn’t available everywhere, that doesn’t mean it shouldn’t be integrated.
It only has to be a small button in the top right of the app (where the location selector and settings icons are), and if coverage isn’t available, that’s ok, it’s not a huge interruption.
Either way that  certainly beats having radar coverage _nowhere_.

Edit: It could also be toggle-able in the sources menu on a per-location basis (or even opt-in!), if that’s still too much of a concern.

**Reply by LinuxinaBit** [reply]
Also **none** of the other sources listed in [RADAR.md](https://github.com/breezy-weather/breezy-weather/blob/main/docs/RADAR.md) have global radar coverage because that is physically impossible; many regions simply don’t have radar coverage _full stop_.

<img width="1080" alt="image" src="https://github.com/user-attachments/assets/b2ebda91-7dd1-4a1f-91bc-f3083334ee50">

_Only the light grey areas have radar coverage._

This is not a reason to avoid including a radar view, and it is not at all RainViewer’s fault.

**Comment by papjul** [comment]
> I find a few problems with that notion, the first being that Breezy Weather displays nothing but freely available weather data that _also_ can be accessed though other sources, and I can’t really figure out what makes displaying weather radar data _specifically_ any different.

There is some confusion here. In both cases, we ask for free APIs. The radar you see on websites is free to look at when you go at their website, but is paid if you want an API access.

The other difference is the display. While the data we display from API is also available on their website, they each have a different presentation. Breezy Weather makes all sources have a unique presentation indeed. Radars, instead, all have very similar looks. It's a map with an overlay. You're not going to get confused/miss information because of a different display in this case.

> not having a radar makes it pretty unusable in that context

Which context requires you to have a radar? "Radar" can mean a lot of things. RainViewer radar is not the same as Windy radar, and not the same as a real-time thunder radar.

If Breezy Weather is *unusable* without a radar, then you only need a radar and you don't need Breezy Weather.

> I don’t _want_ to just use a website to access a weather radar, I want an integrated app that fulfills all my weather checking needs.

Why? Even if a radar was integrated, you would still need an additional tap to access it. Same as having a bookmark/PWA on your homescreen. There is absolute nothing forcing you to have an app doing everything. My philosophy is the opposite: I prefer apps that do one task and do it right.

> This actually seems to be a common request, as there have been many others before me asking for the exact same thing:
> #23
> #767
> #903
> #1178
> #1187
> #1288
> They are all well aware of other radar viewing services being available, but that’s not what these are requests for.
> There is a genuine need for a fully integrated, one-stop-shop FOSS weather app, and if Breezy Weather added a radar, I think it would finally be able to provide that.

I'm not disputing that there is a need. I'm just saying there is no way to offer users a better solution with the available free APIs than what they can have on websites directly. I'm just challenging the "fully integrated" part. Nothing requires that, and it is perfectly acceptable to have an additional icon on your homescreen given the current state (I'm not expecting that state to move, given how costly it can be, compared to simple JSON API (what we use for data)).

> P.S. If any integration examples are needed, [QuickWeather](https://f-droid.org/packages/com.ominous.quickweathe/) is a great example for how to do it. It integrates RainViewer, but that’s ok because the real value in adding a weather radar is not that the data is somehow unique, it’s the fact that it is integrated in the app _at all_.

While some users may be happy in only having RainViewer data in the app, many users would like other kind of radar data, which RainViewer is not able to provide. While radars are usually displayed similarly, they can be technically VERY different to process. It requires a non-negligible amount of work to get it integrated and allow to support multiple sources. If we integrate a new feature, we want to do it the right way. Currently, we have no way to offer something as good as [Windy](https://www.windy.com/), for example. So if we implemented something free, users would still need to go to Windy or most of other websites which are way more complete and friendly.

Even if there were many free and complete sources available for us to use, it would still be a lot of work and its priority would be on the bottom of the list, because there is almost always more added value in other pending features (or maintaining existing ones, such as widgets, which are currently in a state of desolation) than in something you can simply add as bookmark.

TL;DR: the reason there is no radar is already summed up in the document you read: "there is no free radar API we can use that can compete with the other free websites"

**Comment by LinuxinaBit** [comment]
> There is some confusion here. In both cases, we ask for free APIs. The radar you see on websites is free to look at when you go at their website, but is paid if you want an API access.

That is fair, but as you know there are free options. It’s the exact same as the OpenWeatherMap vs OpenMeteo thing.

> The other difference is the display. While the data we display from API is also available on their website, they each have a different presentation. Breezy Weather makes all sources have a unique presentation indeed. Radars, instead, all have very similar looks. It's a map with an overlay.

I disagree with this idea; it can be customized a lot, different color schemes, different methods of navigation, etc. Even the base map layer can be customized by using something like [Carto](https://wiki.openstreetmap.org/wiki/OpenStreetMap_Carto) or other map tile providers. Satellite base layers are even possible; something relatively few radar apps actually provide.

> Why? Even if a radar was integrated, you would still need an additional tap to access it. Same as having a bookmark/PWA on your homescreen.

Because in that case it would be easier to not use Breezy Weather at all. I can check all of the same stuff Breezy Weather shows inside of Windy, but I don’t want to do that because I like using Breezy Weather.

A radar button in Breezy Weather would also open the radar at the currently selected location, would take _so much less_ time to load on a mobile data connection, and run a lot faster on my phone.

> I'm not disputing that there is a need. I'm just saying there is no way to offer users a better solution with the available free APIs than what they can have on websites directly.

You’re misunderstanding the need.
We don’t need a _better_ solution, we just want a usable one that is more accessible than opening a website.
If people want to use a website, that’s fine, but providing the feature _at all_ will decrease that need substantially.

That is also no reason not to at least have it in a list of future features. This feels like a case of “letting perfect be the enemy of good enough.”

> While some users may be happy in only having RainViewer data in the app, many users would like other kind of radar data, which RainViewer is not able to provide.

That is **ok**, it is much better than making none of those users happy, and I think you are vastly overestimating the number of people that need something more than a simple weather radar anyway.

> Even if there were many free and complete sources available for us to use, it would still be a lot of work and its priority would be on the bottom of the list, because there is almost always more added value in other pending features (or maintaining existing ones, such as widgets, which are currently in a state of desolation) than in something you can simply add as bookmark.

And finally we get to the real answer. You don’t have to justify the lack of a weather radar by making up excuses or pretending other websites also having radars makes Breezy Weather having one somehow unnecessary.
Adding a weather radar view is genuinely useful, but if Breezy Weather as a project doesn’t have the resources to add one, just say it. People will understand and there’s no need to pretend otherwise.

Best of luck, and thank you for helping build such a wonderful weather app ❤️

**Reply by papjul** [reply]
> That is fair, but as you know there are free options. It’s the exact same as the OpenWeatherMap vs OpenMeteo thing.

When I'm talking about "free", I'm saying "gratis", not "libre". The only "gratis" source I know with decent coverage is RainViewer on a very specific feature (rain in the next hour). Other are national sources with exotic formats (they aren't always customizable layers, sometimes just some generated images).

> I disagree with this idea; it can be customized a lot, different color schemes, different methods of navigation, etc.

Which idea were you reacting at? Are you talking about a feature you would like that is not available on the websites?

> A radar button in Breezy Weather would also open the radar at the currently selected location, would take so much less time to load on a mobile data connection, and run a lot faster on my phone.

On the first point, I'm pretty sure most websites support saving a specific center point in your bookmarks (at least it is possible on Windy).

On the second point, I'm not sure there would be much difference. You would still have to load the tiles and the layers. I would be surprised it's not what takes most of the bandwidth (assuming you have an ad/tracker blocker).

> You’re misunderstanding the need.
> We don’t need a *better* solution, we just want a usable one that is more accessible than opening a website.
> If people want to use a website, that’s fine, but providing the feature *at all* will decrease that need substantially.

Yes, I misunderstand the need, because you didn't explain it. I seem to understand now your needs are to save a bit on mobile data (which I'm pretty sure would be marginal). Regarding accessibility, which settings are you currently missing on the websites? Did you check all websites to see if one supports it?

> That is **ok**, it is much better than making none of those users happy

Let's agree to disagree. Users with radar needs who don't find it in the app will be completely fine adding an extra icon on their homescreen in most cases IMO

> I think you are vastly overestimating the number of people that need something more than a simple weather radar anyway

They can add a bookmark to the simple weather radar website they use. Those with *simple* needs are probably completely fine not using it integrated in an app, precisely because they have *simple* needs. Especially now that I'm getting the feeling that we are talking about very little details between integrated vs website.

> And finally we get to the real answer. You don’t have to justify the lack of a weather radar by making up excuses or pretending other websites also having radars makes Breezy Weather having one somehow unnecessary.

> if Breezy Weather as a project doesn’t have the resources to add one, just say it. People will understand and there’s no need to pretend otherwise.

That is correct we wouldn't have the resources to add this feature *if* there were decent/complete gratis APIs. But in that case, the issue would be "Open to contributions". In the current state, if you come up with a PR implementing RainViewer, I will not merge it.

> Adding a weather radar view is genuinely useful

I agree, a weather radar can be useful to some, and particularly useful to you as I can see, but I disagree it *has* to be provided by a weather app. A weather app can be good, and let other apps/websites do better work on the radar front, especially if it can't do as good/better.

Being more on the Unix philosophy approach (do one thing and do it well, rather than complicate programs by adding new features), I would even say a better approach would be to make a separate app for that (something you would call Breezy Radar or whatever).

> Best of luck, and thank you for helping build such a wonderful weather app ❤️

Thanks, it's the work of many! ;)

---

You can continue the discussion, but what would unlock the situation is outside our control, so don't take it personally if I'm being late to answer/not giving answers anymore ;)
